burn-import/src/pytorch/converter.rs

```diff
@@ -0,0 +1,250 @@
+use std::{
+    collections::HashMap,
+    env,
+    fs::create_dir_all,
+    path::{Path, PathBuf},
+};
+
+use regex::Regex;
+
+use crate::logger::init_log;
+
+use candle_core::{pickle, Tensor as CandleTensor};
+use clap::ValueEnum;
+
+/// Type of the record to be saved.
+///
+/// Only the record types that support name/value for struct fields can be used.
+/// Positional record types are not supported, such as Bincode, because the order of the fields
+/// dot not match the order of the fields of Pytorch's modules.
+///
+#[derive(Debug, Clone, Default, Copy, ValueEnum)]
+// #[strum(ascii_case_insensitive)]
+pub enum RecordType {
+    /// Pretty JSON format (useful for debugging).
+    // #[strum(serialize = "json")]
+    PrettyJson,
+
+    /// Compressed Named MessagePack.
+    ///
+    /// Note: This may cause infinite build.
+    ///       See [#952 bug](https://github.com/tracel-ai/burn/issues/952).
+    NamedMpkGz,
+
+    /// Uncompressed Named MessagePack.
+    #[default]
+    NamedMpk,
+}
+
+/// Convert states from `.pt` or `.pth` files and save them to the `out_dir`.
+#[derive(Debug, Default)]
+pub struct Converter {
+    out_dir: Option<PathBuf>,
+    /// List of torch files to generate source code from.
+    inputs: Vec<PathBuf>,
+    development: bool,
+    pub(super) half_precision: bool,
+    pub(super) record_type: RecordType,
+    pub(super) key_remap: Vec<(Regex, String)>,
+    pub(super) tagged_modules: Vec<(Regex, ModuleType)>,
+    pub(super) guess_module_type: bool,
+}
+
+/// Type of the module used to tag incoming tensors for automatic conversion to burn module structures.
+///
+/// These modules have weights and their names/shapes need to be converted to burn's structure.
+#[derive(Debug, Clone)]
+pub enum ModuleType {
+    BatchNorm,
+    Conv1d,
+    Conv2d,
+    ConvTranspose1d,
+    ConvTranspose2d,
+    Embedding,
+    GroupNorm,
+    LayerNorm,
+    Linear,
+}
+
+impl Converter {
```

Comment:
I would return `Self` and  `mut self` as input parameter in setters. For example:

```rust
pub fn out_dir(mut self, out_dir: &str) -> Self { ... }
```